File: backend/doisag/src/main/java/dev/uffs/doisag/dto/PasswordRules.java
package dev.uffs.doisag.dto;

// regra de senha forte de todo lugar q cria ou troca senha (RF02.1)
// a tela mostra a mesma regra enquanto a pessoa digita
public final class PasswordRules {

    // de 8 a 64 caracteres com pelo menos uma letra maiuscula um numero e um simbolo
    // simbolo eh tudo q n eh letra nem numero e letra com acento conta como letra
    public static final String PATTERN = "^(?=.*\\p{Lu})(?=.*\\d)(?=.*[^\\p{L}\\p{N}]).{8,64}$";

    public static final String MESSAGE =
            "A senha precisa ter de 8 a 64 caracteres, com pelo menos uma letra maiúscula, um número e um símbolo";

    private PasswordRules() {
    }
}

File: frontend/src/components/form/password-rules.js
// as mesmas regras de senha da api
// a tela marca cada uma conforme a pessoa digita
export const PASSWORD_RULES = [
    {
        id: "length",
        label: "De 8 a 64 caracteres",
        isMet: (password) => password.length >= 8 && password.length <= 64,
    },
    {
        id: "uppercase",
        label: "Uma letra maiúscula",
        isMet: (password) => /\p{Lu}/u.test(password),
    },
    {
        id: "number",
        label: "Um número",
        isMet: (password) => /\d/.test(password),
    },
    {
        id: "symbol",
        label: "Um símbolo, como ! @ # ou -",
        isMet: (password) => /[^\p{L}\p{N}]/u.test(password),
    },
];

export function isStrongPassword(password) {
    return PASSWORD_RULES.every((rule) => rule.isMet(password));
}
